packages/http-proxy-agent/CHANGELOG.md

```diff
@@ -13,6 +13,51 @@
 ### Major Changes
 
 - d99a7c8: Major version bump for all packages
+  - ⚠️ This is a breaking change! The `HttpProxyAgent` constructor argument has been split into two arguments.
+
+#### Upgrading from 5.x to 6.x
+
+In version 5.x, the `HttpProxyAgent` constructor took a single argument of either (A) a `string`, or (B) an object matching the output of
+the [deprecated `url.parse()` method](https://nodejs.org/docs/latest-v14.x/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost)
+_and_ various extra options.
+    
+Now the constructor takes two _separate_ arguments:
+
+* Argument 1: Either (A) a `string`, or (B) a [WHATWG `URL` object](https://nodejs.org/docs/latest-v14.x/api/url.html#url_the_whatwg_url_api)
+* Argument 2 (optional): An object with standard [`http.Agent`](https://nodejs.org/docs/latest-v14.x/api/url.html#url_the_whatwg_url_api),
+  `net.TcpNetConnectOpts`, and `tls.ConnectionOptions` properties.
+
+If you were using an object argument in 5.x, you'll need to change the first argument to match the structure of the `URL` class, and move
+any other options to the second argument.
+
+5.x usage:
+
+```ts
+const agent = new HttpProxyAgent({
+  protocol: 'https:',
+  host: 'myproxy.mydomain.com'
+  port: '1234',
+  auth: 'proxyUser:proxyPass',
+  timeout: 1000
+});
+```
+
+Updated 6.x usage:
+
+```ts
+const agent = new HttpProxyAgent(
+  {
```

Comment:
I suggest just using a string for the first argument for these examples.

Reply:
Well, if someone was using a string argument, then they don't need to change anything, right?
They only need to change something if they used an object argument. That's what I was trying to communicate here, but maybe I need to make that more explicit?

Reply:
I see, makes sense 👍